core/domain/change_domain.py

```diff
@@ -207,15 +213,16 @@ def validate_dict(self, change_dict: Dict[str, Any]) -> None:
         valid_cmd_attribute_specs.pop('name', None)  # type: ignore[misc]
 
         actual_cmd_attributes = copy.deepcopy(change_dict)
-        actual_cmd_attributes.pop('cmd', None)
+        # Here, `actual_cmd_attributes` is of type Mapping and Mapping does not
+        # contain extra methods (e.g: .pop()). But here we are accessing `pop()`
+        # method, which causes MyPy to throw error. Thus to avoid the error,
+        # we used ignore here.
+        actual_cmd_attributes.pop('cmd', None)  # type: ignore[attr-defined]
```

Comment:
What if you replace this with `del actual_cmd_attributes['cmd']` instead of pop?

In general try to find ways to get rid of type-ignores (ditto above for the ignore[misc] too) and Any's.

Reply:
`del actual_cmd_attributes['cmd']` also not work here. Because as mentioned in the comment `Mapping` does not contain extra methods and internally `del` calls `__delitem__` to delete the element.

Also, due to the absence of extra methods this type can be considered a `read-only` type just like `Sequence`.

Reply:
> In general try to find ways to get rid of type-ignores (ditto above for the ignore[misc] too) and Any's.

I'll keep that in mind from now on!!

Reply:
Ah, ok, re del. Thanks for the explanation -- it is a bit weird to me that Dict[str, str], which is editable, is a valid subtype of Mapping[str, Union[str, int]], which is read-only, but ... ¯\_(ツ)_/¯

Reply:
![Screenshot from 2022-06-28 11-29-23](https://user-images.githubusercontent.com/12553576/176104903-56e229ca-acea-4e5a-8c5f-bfb691b3a4bf.png)
Yeah, I'm also surprised at first but after reading the docs I'm confident that we heading in the right direction.